src/test/java/org/jabref/gui/search/GlobalSearchBarTest.java

```diff
@@ -0,0 +1,156 @@
+package org.jabref.gui.search;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+import javafx.scene.Scene;
+import javafx.scene.control.TextInputControl;
+import javafx.scene.layout.HBox;
+import javafx.stage.Stage;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.undo.CountingUndoManager;
+import org.jabref.gui.util.DefaultTaskExecutor;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.search.rules.SearchRules;
+import org.jabref.preferences.PreferencesService;
+import org.jabref.preferences.SearchPreferences;
+import org.jabref.testutils.category.GUITest;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.testfx.api.FxRobot;
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.framework.junit5.Start;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * The GlobalSearchBarTest class currently serves the GUI testing regarding
+ * search history recording improvement. Designed in TestFX testing framework
```

Comment:
Remove "improvement". A test tests current program - not talking about past and future.

I think, the comment on FxRobot is not necessary, because whole JabRef UI test relies on FxRobot. 

You can write documentation to https://github.com/JabRef/jabref/blob/main/docs/code-howtos/testing.md if you want.

In case you want to compare FxRobot with other possibilities, you can write an architecture-decision-record (https://github.com/JabRef/jabref/tree/main/docs/decisions) if you want.

Reply:
I think, the whole JavaDoc comment can be deleted. The content can be derived from a) the class name and b) from general knowledge of JabRef testing.